astropy/modeling/blackbody.py

```diff
@@ -0,0 +1,95 @@
+from __future__ import (absolute_import, unicode_literals, division,
+                        print_function)
+
+from collections import OrderedDict
+
+import numpy as np
+from .core import Fittable1DModel
+from .parameters import Parameter
+from .. import units as u
+
+__all__ = ['BlackBody1D']
+
+
+class BlackBody1D(Fittable1DModel):
+    """
+    One dimensional blackbody model.
+
+    Parameters
+    ----------
+    temperature : :class:`~astropy.units.Quantity`
+        Blackbody temperature
+    bolometric_flux : :class:`~astropy.units.Quantity`
+        The bolometric flux of the blackbody (i.e. the integral over the
+        spectral axis)
+    """
+
+    # We parametrize this model with a temperature and a bolometric flux. The
+    # bolometric flux is the integral of the model over the spectral axis. This
+    # is more useful than simply having an amplitude parameter.
+    temperature = Parameter(default=5000, min=0, unit=u.K)
+    bolometric_flux = Parameter(default=1, unit=u.erg / u.cm ** 2 / u.s)
+
+    # We allow values without units to be passed when evaluating the model, and
+    # in this case the input x values are assumed to be frequencies in Hz.
+    input_units_allow_dimensionless = True
+
+    def evaluate(self, x, temperature, bolometric_flux):
+        """Evaluate the model.
+
+        Parameters
+        ----------
+        x : float or `~numpy.ndarray` or `~astropy.units.Quantity`
+            Frquency at which to compute the blackbody. If no units are given,
+            this defaults to Hz.
+
+        temperature : float or `~numpy.ndarray` or `~astropy.units.Quantity`
+            Temperature of the blackbody. If no units are given, this defaults
+            to Kelvin.
+
+        bolometric_flux : float or `~numpy.ndarray` or `~astropy.units.Quantity`
+            Desired integral for the blackbody.
+
+        Returns
+        -------
+        y : number or ndarray
+            Blackbody spectrum. The units are determined from the units of
+            ``bolometric_flux``.
+        """
+
+        from ..constants import sigma_sb
+        from ..analytic_functions.blackbody import blackbody_nu
```

Comment:
I think we can wait to move `blackbody_nu` over in a separate PR to deprecate `analytic_functions`.